packages/api/lib/writeRecords/write-granules.js

```diff
@@ -399,68 +382,175 @@ const _writeGranule = async ({
-    granulePgRecord: pgGranule,
-    knexOrTransaction: knex,
+/**
+ * Write a granule record to DynamoDB and PostgreSQL
+ *
+ * @param {Object}          params
+ * @param {Object}          params.apiGranuleRecord - Api Granule object to write to the database
+ * @param {number}          params.executionCumulusId - Execution ID the granule was written from
+ * @param {Object}          params.esClient - Elasticsearch client
+ * @param {Object}          params.granuleModel - Instance of DynamoDB granule model
+ * @param {Object}          params.granulePgModel - @cumulus/db compatible granule module instance
+ * @param {Knex}            params.knex - Knex object
+ * @param {Object}          params.postgresGranuleRecord - PostgreSQL granule record to write
+ *                                                         to database
+ * @param {string}          params.snsEventType - SNS Event Type
+ * returns {Promise}
+ */
+const _writeGranule = async ({
+  postgresGranuleRecord,
+  apiGranuleRecord,
+  esClient,
+  executionCumulusId,
+  granuleModel,
+  granulePgModel,
+  knex,
+  snsEventType,
+}) => {
+  const pgGranule = await _writeGranuleRecords({
+    postgresGranuleRecord,
+    apiGranuleRecord,
+    knex,
+    esClient,
+    granuleModel,
+    executionCumulusId,
+    granulePgModel,
+  });
+
+  await _writePostgresFilesFromApiGranuleFiles({
+    apiGranuleRecord,
+    granuleCumulusId: pgGranule.cumulus_id,
+    knex,
+    snsEventType,
+  });
+
+  await _publishPostgresGranuleUpdateToSns({
+    snsEventType,
+    pgGranule,
+    knex,
+  });
+};
+
+/**
+* Method to facilitate parital granule record updates
+* @summary In cases where a full API record is not passed, but partial/tangential updates to granule
+*          records are called for, updates to files records are not required and pre-write
+*          calculation in methods like write/update GranulesFromApi result in unneded
+*          evaluation/database writes /etc. This method updates the postgres/Dynamo/ES datastore and
+*          publishes the SNS update event without incurring unneded overhead.
+* @param {Object}          params
+* @param {Object}          params.apiGranuleRecord - Api Granule object to write to the database
+* @param {number}          params.executionCumulusId - Execution ID the granule was written from
+* @param {Object}          params.esClient - Elasticsearch client
+* @param {Object}          params.granuleModel - Instance of DynamoDB granule model
+* @param {Object}          params.granulePgModel - @cumulus/db compatible granule module instance
+* @param {Knex}            params.knex - Knex object
+* @param {Object}          params.postgresGranuleRecord - PostgreSQL granule record to write
+*                                                         to database
+* @param {string}          params.snsEventType - SNS Event Type
+* @returns {Promise}
+*/
+const updateGranuleRecordAndPublishSns = async ({
```

Comment:
Nit: I would prefer to name the function `writeGranuleRecordAndPublishSns`. `update` implies a different type of operation (only changing certain field values), whereas we are overwriting the records entirely

Reply:
That's partially true, however the intent here is *not* to *create* 'full' records, as it explicitly excludes file writes.    The complexity there is an artifact of the original implementation of this feature and more critically in the difference between dynamo ('writing' a granule record *includes* files).       Given the direction of phase 3, I'm inclined to take your suggestion and ignore the inconsistency in naming in favor of the docstring comment.

7b2a39615